streaming/base/storage/download.py

```diff
@@ -215,11 +221,11 @@ def _gcs_with_service_account(local: str, obj: urllib.parse.ParseResult) -> None
         local (str): Local path (local filesystem).
         obj (ParseResult): ParseResult object of remote path (GCS).
     """
+    from google.auth import default as default_auth
     from google.cloud.storage import Blob, Bucket, Client
 
-    service_account_path = os.environ['GOOGLE_APPLICATION_CREDENTIALS']
-    gcs_client = Client.from_service_account_json(service_account_path)
-
+    credentials, _ = default_auth()
```

Comment:
Does this `default_auth()` checks if a user has set an env var `GOOGLE_APPLICATION_CREDENTIALS`  ? Looks like it is based on [this doc](https://cloud.google.com/docs/authentication/application-default-credentials).

Reply:
It does. There are several `checkers` defined in [`_default`](https://github.com/googleapis/google-auth-library-python/blob/main/google/auth/_default.py#L645-L654), the first of which is [`_get_explicit_environ_credentials`](https://github.com/googleapis/google-auth-library-python/blob/main/google/auth/_default.py#L247). 

That's looking for the path to an `explicit_file` using the environment variable

```
explicit_file = os.environ.get(environment_vars.CREDENTIALS)
```

which is [defined as
](https://github.com/googleapis/google-auth-library-python/blob/f19f610ce6afeea74fd634c356abb24192a843b8/google/auth/environment_vars.py#L36)
```
CREDENTIALS = "GOOGLE_APPLICATION_CREDENTIALS"
```